x/ecocredit/client/tx.go

```diff
@@ -131,84 +125,102 @@ regen tx ecocredit create-class regen1el...xmgqelsw,regen2tl...xsgqdlhy C "metad
+		Use:   "create-project [class-id] [jurisdiction] [metadata] [flags]",
+		Short: "Create a new project within a credit class",
+		Long: `Create a new project within a credit class.
+		
+Parameters:
+
+- class-id:      the ID of the credit class
+- jurisdiction:  the jurisdiction of the project
+- metadata:      any arbitrary metadata to attach to the project
 
-Required Flags:
-  project_id: id of the project
-  issuances:  the amount of issuances to generate
-  start-date: The beginning of the period during which this credit batch was
-              quantified and verified. Format: yyyy-mm-dd.
-  end-date:   The end of the period during which this credit batch was
-              quantified and verified. Format: yyyy-mm-dd.
-  metadata:   issuance metadata
-  issuer:     account address of the batch issuer
-			  `,
-		Args: cobra.ExactArgs(0),
+Optional Flags:
+
+- reference-id: a reference ID for the project`,
+		Example: `regen tx ecocredit create-project C01 "US-WA 98225" regen:13toVgf5UjYBz6J29x28pLQyjKz5FpcW3f4bT5uRKGxGREWGKjEdXYG.rdf
+regen tx ecocredit create-project C01 "US-WA 98225" regen:13toVgf5UjYBz6J29x28pLQyjKz5FpcW3f4bT5uRKGxGREWGKjEdXYG.rdf  --reference-id VCS-001`,
+		Args: cobra.ExactArgs(3),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			projectId, err := cmd.Flags().GetString(FlagProjectId)
+			clientCtx, err := sdkclient.GetClientTxContext(cmd)
 			if err != nil {
 				return err
 			}
 
-			templateIssuance := &core.BatchIssuance{
-				Recipient:              "recipient-address",
-				TradableAmount:         "tradable-amount",
-				RetiredAmount:          "retired-amount",
-				RetirementJurisdiction: "retirement-jurisdiction",
-			}
-
-			numIssuances, err := cmd.Flags().GetUint32(FlagIssuances)
+			referenceId, err := cmd.Flags().GetString(FlagReferenceId)
 			if err != nil {
 				return err
 			}
 
-			issuances := make([]*core.BatchIssuance, numIssuances)
-			for i := range issuances {
-				issuances[i] = templateIssuance
+			msg := core.MsgCreateProject{
+				Admin:        clientCtx.GetFromAddress().String(),
+				ClassId:      args[0],
+				Jurisdiction: args[1],
+				Metadata:     args[2],
+				ReferenceId:  referenceId,
 			}
 
-			startDateStr, err := cmd.Flags().GetString(FlagStartDate)
-			if err != nil {
-				return err
-			}
-			startDate, err := types.ParseDate("start_date", startDateStr)
-			if err != nil {
-				return err
-			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+
+		},
+	}
+
+	cmd.Flags().String(FlagReferenceId, "", "a reference ID for the project")
 
-			endDateStr, err := cmd.Flags().GetString(FlagEndDate)
+	return txFlags(cmd)
+}
+
+// TxGenBatchJSONCmd returns a transaction command that generates JSON to
+// represent a new credit batch.
+func TxGenBatchJSONCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   `gen-batch-json [issuer] [project-id] [issuance-count] [metadata] [start-date] [end-date]`,
```

Comment:
Not sure how valuable this command is given we provide example JSON in `create-batch`. I switched this to arguments rather than flags but maybe flags were being used because of the number of arguments. Although all are required and to be consistent with how we are using flags, arguments made more sense to me.